[RFC] Joomla Dialog, implementation for Media, User Fields

dgrammatiko opened the pull request:
Pull Request for Issue # .

### Summary of Changes

- uses the Joomla Dialog from #40150 for the fields User and Media

### Testing Instructions

- Apply the patch
- Run `npm ci`
- Check that intro, full text images and author fields are working as before

### Actual result BEFORE applying this Pull Request



### Expected result AFTER applying this Pull Request

User Field
![Screenshot 2023-03-20 at 11 33 33](https://user-images.githubusercontent.com/3889375/226314894-58bb3a57-76c3-401a-ae7d-e38a3cf6069e.png)

Media Field
![Screenshot 2023-03-20 at 11 33 48](https://user-images.githubusercontent.com/3889375/226314983-38f7989b-5db7-49cb-b787-abe043861b58.png)


### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

dgrammatiko committed:
Media Field

dgrammatiko committed:
User Field

dgrammatiko committed:
remove the BS PHP

dgrammatiko committed:
cleanup

dgrammatiko committed:
cs

Quy commented:
![users-dialog](https://user-images.githubusercontent.com/368084/226218508-0756c3c2-def7-4224-8e45-0b4766ceed34.png)

Fedik commented on build/media_source/system/js/fields/joomla-field-media.w-c.es6.js:
```diff
@@ -154,16 +137,30 @@ class JoomlaFieldMedia extends HTMLElement {
+    // eslint-disable-next-line
+    const dialog = new JoomlaDialog({
+      popupType: 'iframe',
+      textHeader: Joomla.Text._('JLIB_FORM_CHANGE_IMAGE'),
```
Add `iconHeader: 'icon-pictures'` will make header litle diferent :)
For user field `iconHeader: 'icon-address'`

dgrammatiko replied:
done

Fedik commented:
Looks good, I still need to test.
Couple notes: 
I call the asset "dialog" instead of joomla-dialog, because, well, it joomla already, like we have 'core', 'showon' etc ;)
I would prefer do not to have an extra css, it is very basic, I would do it as part of main template.
For width and height, would use default (defined by css) to keep all popup in consistent size, and set it in script when it need to be diferent.

Fedik commented on build/media_source/system/js/fields/joomla-field-media.w-c.es6.js:
```diff
@@ -154,16 +137,30 @@ class JoomlaFieldMedia extends HTMLElement {
+    // Definig your own buttons:
+    dialog.popupButtons = [
+      { label: Joomla.Text._('JSELECT'), onClick: (event) => this.onSelected(event) },
+      { label: Joomla.Text._('JCANCEL'), onClick: () => dialog.close(), className: 'btn btn-outline-danger ms-2' },
```
If you do not plan to reuse the dialog in future then better call `dialog.destroy()`
Or store the dialog in class property, so next time user click "open", it will open previously stored one.

dgrammatiko replied:
done

dgrammatiko committed:
Changes

- Move the src to media/system/ui
- Call the destroy
- change the Header
- rename the registry entry to dialog

dgrammatiko commented:
> I call the asset "dialog" instead of joomla-dialog, because, well, it joomla already, like we have 'core', 'showon' etc ;)

I renamed them. I think it would be a good idea to have a `system/ui` folder and get also all the custom elements from the other repo here. I mean if there's a way to move the docs and the tests there's no reason to have multiple repos

> I would prefer do not to have an extra css, it is very basic, I would do it as part of main template.

I disagree here. The script should provide some basic styling (mind I didn't copy everything from your PR, only the CE parts) and also the template has the ability to override the assets. Also we're doing that with the alerts and tabs..

> For width and height, would use default (defined by css) to keep all popup in consistent size, and set it in script when it need to be diferent.

Removed them

dgrammatiko committed:
Remove sizing

dgrammatiko committed:
Add the top close button

dgrammatiko committed:
danggling comma

Fedik commented on build/media_source/system/js/fields/joomla-field-media.w-c.es6.js:
```diff
@@ -154,16 +137,33 @@ class JoomlaFieldMedia extends HTMLElement {
+    this.dialog = new JoomlaDialog({
+      popupType: 'iframe',
+      textHeader: Joomla.Text._('JLIB_FORM_CHANGE_IMAGE'),
+      iconHeader: 'icon-address',
```
you mixed icon name ;)
for media field: `'icon-pictures'`
for user field: `'icon-address'`

Not much important, just a thing that can be done.

dgrammatiko replied:
Now that I'm thinking about it this is not ideal, because it requires icon fonts which are something that the CMS should divert from (I know it's ok for the admin but the scripts are also supposed to work on the front end, so it forces the use of font icons universally). Maybe just drop this options and leave only the title and the button...

Fedik replied:
If there no icon font then it will be empty, all fine I think. It basicaly as in every other places in CMS.

dgrammatiko replied:
Yeah, let’s keep it simple here

dgrammatiko committed:
Remove the icon
